src/scripts/main.js

```diff
@@ -1 +1,82 @@
 'use strict';
+
+// Select the notification div
+const notificationDiv = document.querySelector('[data-qa="notification"]');
+
+// Helper function to display messages
+function showNotification(message, isError = false) {
+  notificationDiv.textContent = message;
+  notificationDiv.className = isError ? 'error' : 'success';
+  notificationDiv.style.display = 'block';
+
+  // Hide the notification after 3 seconds
+  setTimeout(() => {
+    notificationDiv.style.display = 'none';
+  }, 3000);
+}
+
+// First Promise
+const firstPromise = new Promise((resolve, reject) => {
+  const clickHandler = () => {
+    resolve('First promise was resolved');
+    document.removeEventListener('click', clickHandler);
+  };
+
+  document.addEventListener('click', clickHandler);
+
+  setTimeout(() => {
+    reject(new Error('First promise was rejected'));
+    document.removeEventListener('click', clickHandler);
+  }, 3000);
+});
```

Comment:
There is a potential race condition in the first promise. If the user clicks within 3 seconds, the promise resolves, but the timeout will still execute and attempt to reject the promise. Consider adding a flag to ensure that only one of the resolve or reject handlers is executed.